core/base/src/TObject.cxx

```diff
@@ -56,6 +56,80 @@ Bool_t TObject::fgObjectStat = kTRUE;
 
 ClassImp(TObject);
 
+namespace ROOT {
+namespace Internal {
+
+// Since we want to run the 'delete side effects' test early on,
+// we avoid using TClass/Cling so we have to use a mock class to
+// to estimate the offsets.
+struct MockTObject {
+   virtual ~MockTObject() = default;
+   // Int_t filler[10] = {0};
+   Int_t fBits = 0;
+   Int_t fUniqueID = 0;
+};
```

Comment:
Indeed. I used the mock class to avoid using `TClass/TCling` as initialization time, but we could also make the function a friend of `TObject` (needed as neither `fBits` nor `fUniqueID` are accessible with non-virtual functions.